Tests/SwiftGRPCNIOTests/GRPCChannelHandlerResponseCapturingTestCase.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import NIO
+import NIOHTTP1
+@testable import SwiftGRPCNIO
+import XCTest
+
+internal struct CaseExtractError: Error {
+  let message: String
+}
+
+@discardableResult
+func extractHeaders(_ response: RawGRPCServerResponsePart) throws -> HTTPHeaders {
+  guard case .headers(let headers) = response else {
+    throw CaseExtractError(message: "\(response) did not match .headers")
+  }
+  return headers
+}
+
+@discardableResult
+func extractMessage(_ response: RawGRPCServerResponsePart) throws -> ByteBuffer {
+  guard case .message(let message) = response else {
+    throw CaseExtractError(message: "\(response) did not match .message")
+  }
+  return message
+}
+
+@discardableResult
+func extractStatus(_ response: RawGRPCServerResponsePart) throws -> GRPCStatus {
+  guard case .status(let status) = response else {
+    throw CaseExtractError(message: "\(response) did not match .status")
+  }
+  return status
+}
+
+class CollectingChannelHandler<OutboundIn>: ChannelOutboundHandler {
+  var responses: [OutboundIn] = []
+
+  func write(ctx: ChannelHandlerContext, data: NIOAny, promise: EventLoopPromise<Void>?) {
+    responses.append(unwrapOutboundIn(data))
+  }
+}
+
+class GRPCChannelHandlerResponseCapturingTestCase: XCTestCase {
+  static let echoProvider: [String: CallHandlerProvider] = ["echo.Echo": EchoProvider_NIO()]
+
+  func configureChannel(withHandlers handlers: [ChannelHandler]) -> EventLoopFuture<EmbeddedChannel> {
+    let channel = EmbeddedChannel()
+    return channel.pipeline.addHandlers(handlers, first: true)
+      .map { _ in channel }
+  }
+
+  /// Waits for `count` responses to be collected and then returns them. The test fails if the number
+  /// of collected responses does not match the expected.
+  func waitForGRPCChannelHandlerResponses(
```

Comment:
Would you mind adding a remark that `callback` is called _after_ the channel has been set up, but _before_ this method returns, and that it is meant to contain the calls that actually lead to the channel being "filled" with messages?